Kubernetes/HOWTO-on-prem.md

```diff
@@ -102,91 +75,219 @@ sudo cp -r !(*.tgz) ${CNI_BIN}
 
+    ~/kube $ MASTER_IP=10.123.45.67   # example! replace
     ~/kube $ cd certs
-    ~/kube/certs $ ./generate-certs.sh 10.123.45.67   # example! replace
-
-### Allow Routing ###
-_This step may be optional, depending on whether or not you've configured your intended cluster subnet to be routable already_.
+    ~/kube/certs $ ./generate-certs.sh $MASTER_IP
 
-Windows nodes will each grab a /24 from the cluster CIDR, so it's 192.168.0.0/16, the first Windows node will use 192.168.0.1/24 for its pods. Pass the first two octets of your clister CIDR to generate the routes:
-
-    ~/kube $ generate-routes.sh 192.168
 
 ### Prepare Manifests & Addons ###
 In the `manifest` folder, run the Python script, passing your master IP and the _full_ cluster CIDR:
 
-    $ python2 generate.py 10.123.45.67 192.168.0.0/16
+    $ ./generate.py $MASTER_IP 192.168.0.0/16
+
+This will generate a set of YAML files. You should [re]move the Python script so that Kubernetes doesn't mistake it for a manifest; this will cause problems.
+
+
+### *Optional*: Customize Cluster CIDR ###
+It's possible that you don't have a continous /16 open for cluster routing. This requires more effort on your part as the configurator, as you'll have to explicitly specify per-node CIDRs that you have available. First, you will need to modify the `kube-controller-manager.yaml` configuration file that we just generated. Search for this line:
+
+    - "--allocate-node-cidrs=True"
+
+and change it to `=False`. Now, Kubernetes will not automatically assign subnets to new nodes. Read [this section](#optional-customize-windows-node-cidr) to know what to do on the Windows nodes; the same logic applies to the `start-kubelet.sh` script on the master, and any Linux worker node `kubelet` calls.
 
-This will generate a set of YAML files. You should [re]move the Python script so that K8s doesn't mistake it for a manifest and cause problems.
 
 ### Configure & Run Kubernetes ###
-Run the script `configure-kubectl.sh` passing your external IP again. This will create a configuration in `~/.kube/config` containing all of the certificate data and other parameters:
+Configure Kubernetes with the certificates we generated previously. This will create a configuration at `~/.kube/config`:
+
+    ~/kube $ ./configure-kubectl.sh $MASTER_IP
 
-    ~/kube $ ./configure-kubectl.sh 10.123.45.67
+Finally, we're ready to start `kubelet`, the Kubernetes "client." This script runs indefinitely, so open another shell afterward to keep working:
 
-Run the script `start-kubelet.sh`. You may need to run it with `sudo`. Then, in another shell, watch your working directory (`~/kube`) until the folder `kubelet/` appears. Immediately copy the Kubernetes configuration file to it:
+    ~/kube $ sudo ./start-kubelet.sh
+
+Watch your working directory until the `kubelet` _folder_ appears in it. Immediately copy the Kubernetes configuration file to it:
 
     ~/kube $ sudo cp ~/.kube/config kubelet/

```

Comment:
I still don't get why we have to do this... :-)

Reply:
I wish I knew :P